Source/WebCore/page/Navigation.h

```diff
@@ -195,6 +197,28 @@ class Navigation final : public RefCounted<Navigation>, public EventTarget, publ
     static Vector<Ref<HistoryItem>> filterHistoryItemsForNavigationAPI(Vector<Ref<HistoryItem>>&&, HistoryItem&);
 
 private:
+    // Rate limiter to prevent excessive navigation requests.
+    class NavigationRateLimiter {
```

Comment:
The class name can be simply `RateLimiter` if you define this class as internal class.